src/lib.rs

```diff
@@ -760,54 +752,51 @@ impl StreamDeck {
     /// Sleeps the device
     pub fn sleep(&self) -> Result<(), StreamDeckError> {
         self.initialize()?;
-        match self.kind {
-            Kind::Akp153 | Kind::Akp153E | Kind::Akp153R | Kind::Akp815 | Kind::Akp03E | Kind::Akp03R | Kind::MiraBoxHSV293S => {
-                let mut buf = vec![0x00, 0x43, 0x52, 0x54, 0x00, 0x00, 0x48, 0x41, 0x4e];
 
-                ajazz_extend_packet(&self.kind, &mut buf);
+        if !self.kind.is_mirabox() {
+            return Err(StreamDeckError::UnsupportedOperation);
+        }
 
-                write_data(&self.device, buf.as_slice())?;
+        let mut buf = vec![0x00, 0x43, 0x52, 0x54, 0x00, 0x00, 0x48, 0x41, 0x4e];
 
-                Ok(())
-            }
+        ajazz_extend_packet(&self.kind, &mut buf);
 
-            _ => Err(StreamDeckError::UnsupportedOperation),
-        }
+        write_data(&self.device, buf.as_slice())?;
+
+        Ok(())
     }
 
     /// Make periodic events to the device, to keep it alive
     pub fn keep_alive(&self) -> Result<(), StreamDeckError> {
         self.initialize()?;
-        match self.kind {
-            Kind::Akp153 | Kind::Akp153E | Kind::Akp153R | Kind::Akp815 | Kind::Akp03E | Kind::Akp03R | Kind::MiraBoxHSV293S => {
-                let mut buf = vec![0x00, 0x43, 0x52, 0x54, 0x00, 0x00, 0x43, 0x4F, 0x4E, 0x4E, 0x45, 0x43, 0x54];
-                ajazz_extend_packet(&self.kind, &mut buf);
-                write_data(&self.device, buf.as_slice())?;
-                Ok(())
-            }
 
-            _ => Err(StreamDeckError::UnsupportedOperation),
+        if !self.kind.is_mirabox() {
+            return Err(StreamDeckError::UnsupportedOperation);
         }
+
```

Comment:
Again, an else block would be more Rust-y and match the original code.

Reply:
Not sure about this specific case, an early return for unsupported devices looks alright to me in these functions, and reduces nesting depth

Reply:
Nevermind, you're right, I read it wrong (as you can see I also didn't suggest this on the other ones that looked like this)